### PR TITLE
Support Liquibase changelog parameters

### PR DESCRIPTION
**Pull request by tisoft**
Adds support for setting [liquibase changelog parameters](https://docs.liquibase.com/concepts/basic/changelog-parameters.html).

Usage in `application.properties`:

```
quarkus.liquibase.params."table.prefix"=MY_CUSTOM_PREFIX
quarkus.liquibase.params.someParam=Some Value
```

Would replace the expressions `${table.prefix}` and `${someParam}` with the given values inside the liquibase changelog.

### CODE REVIEW

**Comment by tisoft** [comment]
Usage example:

Using this changelog:

```
<?xml version="1.1" encoding="UTF-8" standalone="no"?>
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">

    <property name="table.prefix" value=""/>

    <changeSet author="dev (generated)" id="id-1">
        <createTable tableName="${table.prefix}TEST">
            <column name="ID" type="VARCHAR(255)">
                <constraints nullable="false"/>
            </column>
            <column name="NAME" type="VARCHAR(255)"/>
        </createTable>
    </changeSet>

</databaseChangeLog>

```

together with a custom hibernate naming strategy:

```
import org.eclipse.microprofile.config.ConfigProvider;
import org.hibernate.boot.model.naming.Identifier;
import org.hibernate.boot.model.naming.PhysicalNamingStrategy;
import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;

public class PrefixNamingStrategy implements PhysicalNamingStrategy {

   private final String tablePrefix;

   public PrefixNamingStrategy () {
      this.tablePrefix = ConfigProvider.getConfig ().getValue ("table.prefix", String.class);
   }

   @Override
   public Identifier toPhysicalCatalogName (Identifier name, JdbcEnvironment jdbcEnvironment) {
      return name;
   }

   @Override
   public Identifier toPhysicalSchemaName (Identifier name, JdbcEnvironment jdbcEnvironment) {
      return name;
   }

   @Override
   public Identifier toPhysicalTableName (Identifier name, JdbcEnvironment jdbcEnvironment) {
      return Identifier.toIdentifier (tablePrefix + name.getText (), name.isQuoted ());
   }

   @Override
   public Identifier toPhysicalSequenceName (Identifier name, JdbcEnvironment jdbcEnvironment) {
      return Identifier.toIdentifier (tablePrefix + name.getText (), name.isQuoted ());
   }

   @Override
   public Identifier toPhysicalColumnName (Identifier name, JdbcEnvironment jdbcEnvironment) {
      return name;
   }
}
```

and the following config:

```
table.prefix=FC_
quarkus.liquibase.database-change-log-lock-table-name=${table.prefix}DATABASECHANGELOGLOCK
quarkus.liquibase.database-change-log-table-name=${table.prefix}DATABASECHANGELOG
quarkus.liquibase.params."table.prefix"=${table.prefix}

quarkus.hibernate-orm.physical-naming-strategy=PrefixNamingStrategy
```

allows me to have a custom table prefix.

I do need this, since my application uses a shared database, and there are overlaps in table names with another application.

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceRuntimeConfig.java [hunk 1]
```diff
@@ -68,6 +70,12 @@ public static final LiquibaseDataSourceRuntimeConfig defaultConfig() {
     @ConfigItem
     public Optional<List<String>> labels = Optional.empty();
 
+    /**
+     * map of parameters, that can be used inside liquibase scripts
+     */
+    @ConfigItem
+    public Map<String, String> params = new HashMap<>();
```
Comment (gsmet) [inline extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseDataSourceRuntimeConfig.java:77]: I would make it `changelogParameters`. `params` is too generic.

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java [hunk 2]
```diff
@@ -64,7 +64,11 @@ public Liquibase createLiquibase() {
                     database.setDefaultSchemaName(config.defaultSchemaName.get());
                 }
             }
-            return new Liquibase(config.changeLog, resourceAccessor, database);
+            Liquibase liquibase = new Liquibase(config.changeLog, resourceAccessor, database);
+            if (config.params != null) {
+                config.params.forEach((k, v) -> liquibase.getChangeLogParameters().set(k, v));
```
Comment (gsmet) [inline extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java:69]: We avoid lambdas in runtime code as they have a cost.

The map should never be null.

**Review by gsmet — CHANGES_REQUESTED** [review]
Interesting, thanks! I added some minor comments.

**Comment by gsmet** [comment]
I'm closing this PR as it shouldn't target 1.8 but master.

I will move the commit to a new PR targeting master and adjust a few things. I will ping you in the PR so you can check that things are ok for you.

**Pull request closed without merging**